jwt/algorithms.py

```diff
@@ -343,7 +343,10 @@ def prepare_key(self, key: AllowedRSAKeys | str | bytes) -> AllowedRSAKeys:
                         RSAPrivateKey, load_pem_private_key(key_bytes, password=None)
                     )
             except ValueError:
-                return cast(RSAPublicKey, load_pem_public_key(key_bytes))
+                try:
+                    return cast(RSAPublicKey, load_pem_public_key(key_bytes))
+                except (ValueError, UnsupportedAlgorithm):
```

Comment:
can we add a test that covers this behavior when using the version of cryptography that introduces the behavior?

Reply:
Sorry it took me a while to get around to this @jpadilla. I've added a test now.